### PR TITLE
Convert to ReScript

### PR DESCRIPTION
**Pull request by alex35mil**
This PR:

- Swaps `bs-platform` with `rescript`
- Swaps `reason-react` with `@rescript/react`
- Converts to ReScript syntax
- Renames package to `rescript-storybook`

### CODE REVIEW

File: example/.gitignore [hunk 1]
```diff
@@ -1,4 +1,4 @@
 node_modules/
 src/*.js
 lib
-yarn.lock
```
Comment (alexfedoseev) [inline example/.gitignore:4]: This file was committed and gitignored. It should be either gitignored and rm'ed from the history or ungitignored and commited. I did the latter.

**Comment by peterpme** [comment]
@alexfedoseev ! Thanks for doing this. I totally missed this! We're in the process of finally upgrading to `@rescript/react` ourselves and will merge this in. Thanks!

**Comment by alexfedoseev** [comment]
Hey @peterpme, how’s it going? One of our teams is having issues with storybook being a git dependency. Any chance you merge and release it anytime soon?

**Comment by peterpme** [comment]
Hey @alexfedoseev! 

Thanks for the reminder, sorry for the delay :) I will release this today

**Comment by peterpme** [comment]
Done @alexfedoseev https://github.com/draftbit/rescript-storybook/releases/tag/v2.0.1

Also, do you want to maintain this? We don't use storybook anymore so I'm a little slow 😅

**Comment by alex35mil** [comment]
Thanks @peterpme!

> Also, do you want to maintain this? We don't use storybook anymore so I'm a little slow 😅

TBH I never was an active storybook user, so I'm not the best person to maintain it. But let me ask my colleagues if somebody wants to keep an eye on it.

**Comment by alex35mil** [comment]
@peterpme I changed my mind 😅 Can you transfer the repo to https://github.com/shakacode ?

**Pull request merged**